src/note.cpp

```diff
@@ -8,24 +8,25 @@
 /*************************************************************************/
 
 #include "note.h"
+#include "CommandSet.hpp"
 #include "Descriptor.hpp"
 #include "buffer.h"
 #include "merc.h"
 #include "string_utils.hpp"
-#include "trie.h"
 
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
+#include <functional>
+#include <memory>
 
 static NOTE_DATA *note_first;
 static NOTE_DATA *note_last;
 
-static void *trie;
-
-typedef void(note_fn_t)(CHAR_DATA *, const char *);
+typedef std::function<void(CHAR_DATA *ch, const char *arg)> note_fn;
```

Comment:
`typedef` is the Old Way™. It still works (as everything does in C++), but it's nicer to use:

```cpp
using note_fn = std::function<void(CHAR_DATA *ch, const char *arg)>;
```

(mainly as that syntax works in more cases)

Reply:
right, and we already have that defined in interp

Reply:
Done.